Spec/RealtimeClient.swift

```diff
@@ -352,6 +352,478 @@ class RealtimeClient: QuickSpec {
+                        }
+                    }
+
+                    expect(client.auth.tokenDetails?.token).toNot(equal(testToken))
+
+                    guard let transport = client.transport as? TestProxyTransport else {
+                        fail("TestProxyTransport is not set"); return
+                    }
+
+                    expect(transport.protocolMessagesReceived.filter{ $0.action == .Disconnected }).to(beEmpty())
+                    expect(transport.protocolMessagesReceived.filter{ $0.action == .Error }).to(beEmpty())
+                }
+
+                // RTC8a1 - part 3
+                it("when capabilities are downgraded, client should receive an ERROR ProtocolMessage with a channel property") {
+                    let options = AblyTests.commonAppSetup()
+                    options.autoConnect = false
+                    let testToken = getTestToken()
+                    options.token = testToken
+                    let client = ARTRealtime(options: options)
+                    defer { client.dispose(); client.close() }
+                    client.setTransportClass(TestProxyTransport.self)
+
+                    let channel = client.channels.get("test")
+                    waitUntil(timeout: testTimeout) { done in
+                        client.connect()
+                        channel.attach() { _ in
+                            done()
+                        }
+                    }
+
+                    waitUntil(timeout: testTimeout) { done in
+                        let partialDone = AblyTests.splitDone(2, done: done)
+
+                        channel.once(.Failed) { error in
+                            guard let error = error else {
+                                fail("ErrorInfo is nil"); partialDone(); return
+                            }
+                            expect(error).to(beIdenticalTo(channel.errorReason))
+                            expect(error.code).to(equal(40160))
+
+                            guard let transport = client.transport as? TestProxyTransport else {
+                                fail("TestProxyTransport is not set"); partialDone(); return
+                            }
+
+                            let errorMessages = transport.protocolMessagesReceived.filter{ $0.action == .Error }
+                            expect(errorMessages).to(haveCount(1))
+
+                            guard let errorMessage = errorMessages.first else {
+                                fail("Missing ERROR protocol message"); partialDone(); return
+                            }
+                            expect(errorMessage.channel).to(contain("test"))
+                            expect(errorMessage.error?.code).to(equal(error.code))
+                            partialDone()
+                        }
+
+                        let tokenParams = ARTTokenParams()
+                        tokenParams.capability = "{\"test\":[\"subscribe\"]}"
```

Comment:
If the new token capability provides access to `test`, and you are attached to `test`, this should not fail?  Something appears to be wrong here.